Detections/3pNetworkDetetions/PhishiglinkExecutionObserved.yaml

```diff
@@ -0,0 +1,83 @@
+id: 2fed0668-6d43-4c78-87e6-510f96f12145
+name: Phishing link click observed in 3P Network device
+description: |
+  'sucessfull Phishing link accessed via "Palo Alto Networks","Fortinet","Check Point","Zscaler" devices.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId:  OfficeATP
+    dataTypes:
+    - OfficeATP
+    - Office365
+    - CommonSecurityLog
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CommandAndControl
+relevantTechniques:
+  - T1071
+query: |
+    //Finding security alerts and extracting the Entities 
+    SecurityAlert
+    //extracting Alert Entities 
+    | extend Entities = parse_json(Entities)
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'account'
+    | extend EntityUPN = iff(isempty(Entity.UserPrincipalName), tostring(strcat(Entity.Name, "@", tostring (Entity.UPNSuffix))), tostring(Entity.UserPrincipalName))
+    )
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'ip'|extend EntityIp = tostring(Entity.Address)
+    )
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'host'| extend EntityHost = tostring(Entity.HostName)
+    )
+    | mv-apply Entity = Entities on
+    (
+    where Entity.Type == 'url'
+    | extend EntityUrl = tostring(Entity.Url)
+    )
+    | where not(ipv4_is_private(EntityIp))
+    | summarize
+    AccountUpn=take_any(EntityUPN),IPAddress=take_any(EntityIp),HostName=take_any(EntityHost),Url=take_any(EntityUrl),AlertTime= min(TimeGenerated)by SystemAlertId, ProductName
+    // filtering 3pnetwork devices 
+    | join kind= inner (CommonSecurityLog
+    | where DeviceVendor has_any  ("Palo Alto Networks", "Fortinet", "Check Point", "Zscaler")
+    | project
+    3plogTime=TimeGenerated,
+    DeviceVendor,
+    DeviceProduct,
+    Activity,
+    DestinationHostName,
+    DestinationIP,
+    RequestURL,
+    MaliciousIP,
+    SourceUserName,
+    IndicatorThreatType,
+    ThreatSeverity,
+    ThreatConfidence)on $left.Url == $right.RequestURL
+    // Applied the condition where alert trigger 1st and then the 3p Network activity execution 
+    | where AlertTime > 3plogTime   	
+entityMappings:
+    - entityType: Account
+      fieldMappings:
+      - identifier: FullName
```

Comment:
We need to use the Name and UPNSuffix for Account to get a strong identifier - more details are here: https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide#entitymappings

Reply:
addressed